pipeline.go

```diff
@@ -33,26 +34,44 @@ type Pipeline struct {
 
 // NewPipeline creates a new Pipeline from given files which ignores the
 // existence of steps with names provided in ignoreSteps.
-func NewPipeline(definitionPath, environmentPath string, ignoredSteps types.StringSet) (*Pipeline, error) {
+func NewPipeline(definitionPath, environmentPath string, environment types.MappingWithEquals, ignoredSteps types.StringSet) (*Pipeline, error) {
 	p := &Pipeline{}
-	def, err := NewPipelineDefinition(definitionPath, ignoredSteps)
+	// Load environment
+	env, _ := NewPipelineEnvironment(environmentPath, environment, ignoredSteps)
+	p.Environment = env
```

Comment:
I think this can be written
`p.Environment, _ = NewPipelineEnvironment(environmentPath, environment, ignoredSteps)` also, why can you ignore the error here?

Reply:
We can ignore errors as a sane default is always used, but I'll add some handling.